src/guildenstern/dispatcher.nim

```diff
@@ -20,7 +20,8 @@ proc process(gs: ptr GuildenServer, fd: posix.SocketHandle, data: ptr SocketData
     if gs.selector.contains(fd):
       try: gs.selector.updateHandle(fd, {Event.Read})
       except: echo "addHandle error: " & getCurrentExceptionMsg()
```

Comment:
Calling `fd.close()` here leads to potential double-closing of file descriptor which is dangerous.  Since file descriptors can be allocated/manipulated by other threads, this can essentially clobber another thread/client's socket in a multi-threaded configuration.   It's safer to call `fd.close()` or `closeSocket()` within the context where errors happen.

(Note:  Unfortunately, current nim standard library implementation of socket `close()` does not raise exceptions on double-close, so this went undetected...)

Reply:
BTW, I added the `fulldebug` log statement just to maintain the `else:` branch;  not sure it's really helpful.